plugins/modules/zypper_repository.py

```diff
@@ -443,6 +438,24 @@ def exit_unchanged():
         if 'gpgcheck' in repofile_items:
             repodata['gpgcheck'] = repofile_items['gpgcheck']
 
+    # override boolean parameters in repodata with provided entries in module.params
+    # in the language that zypper lr -x provides for easier comparison
+    if 'enabled' in module.params:
+        if module.params['enabled']:
+            repodata['enabled'] = '1'
+        else:
+            repodata['enabled'] = '0'
+    if 'disable_gpg_check' in module.params:
+        if module.params['disable_gpg_check']:
+            repodata['gpgcheck'] = '0'
+        else:
+            repodata['gpgcheck'] = '1'
+    if 'autorefresh' in module.params:
+        if module.params['autorefresh']:
+            repodata['autorefresh'] = '1'
+        else:
+            repodata['autorefresh'] = '0'
```

Comment:
This check if the key exists, however, is redundant. They will always exist. Please simplify this to:
```suggestion
    repodata['enabled'] = '1' if module.params['enabled'] else '0'
    repodata['gpgcheck'] = '0' if module.params['disable_gpg_check'] else '1'
    repodata['autorefresh'] = '1' if module.params['autorefresh'] else '0'
```

Reply:
I think, it would be more intuitive to remove the default values and then have the additional check for the flags in module.params. 
Then you could do the following:

  - name: Disable the repository
    zypper_repository:
      repo: repo_path
      enabled: false
  - name: Disable the gpg-check
    zypper_repository:
      repo: repo_path
      disable_gpg_check: false

and the repository would still be disabled. Currently - with the default values - the second task would enable it again.

I understand though, that this changes the behaviour of the module (against what was previously declared in the documentation, because actually, the update functionality is broken completely when all data is overridden with what is found on the file system).
On the other hand, I can't picture many situations, where you need two different modifications on the same module in two steps. So I am ok with what you suggested